projects/vdk-plugins/vdk-jobs-troubleshooting/src/vdk/plugin/jobs_troubleshoot/troubleshoot_utilities/healthcheck_server.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from http.server import HTTPServer
+from http.server import SimpleHTTPRequestHandler
+from threading import Thread
+from typing import Any
+
+
+class HealthCheckServer:
+    """
+    A class for creating an HTTP server for serving requests.
+    """
+
+    def __init__(self, port: int, handler: Any = None):
+        """
+        Initializes a new instance of the HealthCheckServer class.
+
+        Parameters:
+            port (int): The port number on which the server will listen for requests.
+            handler (Any, optional): The request handler class. Defaults to SimpleHTTPRequestHandler.
+        """
+
+        if handler:
+            self._server = HTTPServer(("", port), handler)
+        else:
+            self._server = HTTPServer(("", port), SimpleHTTPRequestHandler)
```

Comment:
Why would you start it up without the thread dump handler? what would this do?

Reply:
agreed. added an error log instead.